src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9843,12 +9843,76 @@ private ArgumentSyntax ParseArgumentExpression(bool isIndexer)
                 else
                 {
                     expression = this.ParseSubExpression(Precedence.Expression);
+
+                    // See if the expression is an invocation that could also be successfully parsed and interpreted
+                    // as a deconstruction target. I.e. something like "var (x, y)" or "var (x, (y, z))".
+                    // We should report an error in this case because we plan to support deconstruction for out arguments
+                    // in the future. We need to ensure that we do not successfully interpret this as an invocation of a
+                    // ref-returning method named var with two, or more arguments. 
+                    if (IsDeconstructionCompatibleArgument(refOrOutKeyword, expression))
+                    {
+                        expression = this.AddError(expression, ErrorCode.ERR_OutVarDeconstructionIsNotSupported);
+                    }
                 }
             }
 
             return _syntaxFactory.Argument(nameColon, refOrOutKeyword, expression);
         }
 
+        private static bool IsDeconstructionCompatibleArgument(SyntaxToken refOrOutKeyword, ExpressionSyntax expression)
+        {
+            if (refOrOutKeyword?.Kind == SyntaxKind.OutKeyword &&
+                expression.Kind == SyntaxKind.InvocationExpression)
+            {
+                var invocation = (InvocationExpressionSyntax)expression;
+                ExpressionSyntax invocationTarget = invocation.Expression;
+
+                return invocationTarget.Kind == SyntaxKind.IdentifierName &&
```

Comment:
>invocationTarget [](start = 23, length = 16)

I don't believe this is the language the LDM is likely to land on for this syntactic restriction. Rather, I think the most likely language rule is that you're not allowed to have the out argument be an invocation of a method given by its simple name that is `var`.